### PR TITLE
Instead of python, use PYTHON_BIN_PATH in pip.sh.

### PR DESCRIPTION
**Pull request by gunan**

### CODE REVIEW

**Comment by caisq** [comment]
Help me understand why this change is necessary. The current approach (https://github.com/tensorflow/tensorflow/pull/15217) works. 
`python -m virtualenv` can be used to create virtualenvs containing any desired Python versions, regardless of the Python version of the `python` binary. The desired Python version is specified with the `-p <PATH_TO_PYTHON_BIN>` argument, as in the current approach. For example, if `PATH_TO_PYTHON_BIN` is /usr/bin/python3.6, the created virtualenv will have Python 3.6 inside it, even though the `python` in `python -m virtualenv -p ...` is Python 2.7.

I verified this with the CPU and GPU builds that we have, e.g.,
http://ci.tensorflow.org/view/Experimental/job/experimental-cais-linux-gpu/7/ (the python 3.6 build failure there is unrelated)
http://ci.tensorflow.org/view/Tensorflow%20Jenkins%20Monitored%20builds/job/nightly-matrix-linux-gpu/734/ (The no_pip build failure there is unrelated).

The approach in this PR has the following problems
1) it is less economical because it requires installation of more packages.
2) `python3.6 -m virtualenv ...` probably doesn't work, because `venv` is required instead.

**Comment by gunan** [comment]
I am more comfortable with using something standard. According to this link:
https://stackoverflow.com/questions/41573587/what-is-the-difference-between-venv-pyvenv-pyenv-virtualenv-virtualenvwrappe
`venv` is something that is supposed to be in the core package, but as mentioned in the link "although for some reason some distros separate it out into a separate distro package, such as  python3-venv on Ubuntu/Debian)" And we are using ubuntu.
So it is not guaranteed that venv will always work for us. Which will make things flaky. Once we push our vms to ubuntu 16, we will see flaakes. When we move things to another system, we can see more flakes. Installing virtualenv will just make things stable.

As for the economy. Here is the virtualenv package:
https://pypi.python.org/pypi/virtualenv
The size is around 1 MB. docker will compress the images, and this will account to a very small amount.
Moreover, for the docker images this script affects the economy hit is a non-issue because we only use this in CI docker images.

**Comment by caisq** [comment]
@gunan, i'm not sure how the comment ""although for some reason some distros separate it out into a separate distro package, such as python3-venv on Ubuntu/Debian)" is relevant here. The comment is about the fact that in some Ubuntu/Debian distros, `venv` is not available in the default python **3** install, but need to be installed separately with the `python3-venv` deb package. This is not relevant to the current working approach of using the virtualenv form python **2** (i.e., the `python-virtualenv` package) to create virtualenvs for whatever target python version desired. `python-virtualenv` is a deb package available on ubuntu 14.04, 16.04, 17.04, and 17.10:

https://packages.ubuntu.com/search?suite=trusty&searchon=names&keywords=python-virtualenv
https://packages.ubuntu.com/search?suite=xenial&searchon=names&keywords=python-virtualenv
https://packages.ubuntu.com/search?suite=zesty&searchon=names&keywords=python-virtualenv
https://packages.ubuntu.com/search?suite=artful&searchon=names&keywords=python-virtualenv

**Comment by gunan** [comment]
Sorry, I realized I missed a key piece of information.
The current approach does not fully work on all our setups:
http://ci.tensorflow.org/view/tf-nightly/job/tf-nightly-linux/

This is blocking our nightly binaries for cuda9, which has been a major ask from nvidia and many users.
So either with this patch, or another I want to roll out a fix today and finally push cuda9 enabled nightlies.

**Comment by gunan** [comment]
The specific failure this patch addresses is here:
http://ci.tensorflow.org/view/tf-nightly/job/tf-nightly-linux/TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=PIP,TF_BUILD_PYTHON_VERSION=PYTHON3.6,label=cpu-slave/150/console

**Comment by caisq** [comment]
Ah - that's unfortunate. Sorry I didn't try the PR with tf-nightly jobs. But I'm still not sure if this PR is going to fix all cases. For example, if `$PYTHON_BIN_PATH` points to a python 3 version, the module name used after `-m` should be `venv`, but not `virtualenv`.

**Comment by gunan** [comment]
I have ran the python2 and 3 builds with this PR.
After we do "pip install virtualenv" `$PYTHON_BIN_PATH -m virtualenv`
command seems to work on my machine on all the cases I tested.
It looks like "venv" is just "commonly together with python3 distros" but
we can always install and use virtualenv.

File: tensorflow/tools/ci_build/builds/pip.sh [hunk 1]
```diff
@@ -299,7 +299,7 @@ create_activate_virtualenv_and_install_tensorflow() {
   # Use the virtualenv from the default python version (i.e., python-virtualenv)
   # to create the virtualenv directory for testing. Use the -p flag to specify
   # the python version inside the to-be-created virtualenv directory.
-  python -m virtualenv -p "${PYTHON_BIN_PATH}" ${VIRTUALENV_FLAGS} \
+  $PYTHON_BIN_PATH -m virtualenv -p "${PYTHON_BIN_PATH}" ${VIRTUALENV_FLAGS} \
```
Comment (caisq) [inline tensorflow/tools/ci_build/builds/pip.sh]: Bash nit: --> `"${PYTHON_BIN_PATH}"` to prevent path splitting.

**Comment by gunan** [comment]
Done,
Sorry, I was looking through the comments, then I realized I misread your first comment, and thought you were aware of the python 3.6 build breakage. My mistake, should have mentioned it in the PR description.

**Review by caisq — APPROVED** [review]
Thanks for the fix!

**Pull request merged**